Release v0.4.6: bump up PACs to 0.33.5 and adjust HAL code

michalfita opened the pull request:
Formality.
After bump up of PACs to 0.33.5 we had to bump MSRV to 1.74, tweak some other dependencies and fix HAL code.

michalfita committed:
Attempt to release v0.4.5 with PACs 0.30.3

The rust-embedded/svd2rust#863 bug prevents this from working.

michalfita committed:
Attempt to v0.4.6 with PACs 0.33.5

martinmortsell commented:
Changes look sound.
```
error: package `zerofrom v0.1.6` cannot be built because it requires rustc 1.81 or newer, while the currently active rustc version is 1.74.0
Either upgrade to rustc 1.81 or newer, or use
cargo update zerofrom@0.1.6 --precise ver
where `ver` is the latest version of `zerofrom` supporting rustc 1.74.0
```
CI breaks because rust version is too low. The formatting check also breaks.

michalfita committed:
Second attempt to v0.4.6 with PACs v0.33.5

michalfita committed:
Commit `Cargo.lock` to stick with dependencies working with rustc 1.74

michalfita committed:
Prevent dead code error on  `afec::Samples`

Why it's unused is a mystery, probably was planned to be returned as multichannel measurement.

michalfita committed:
Fix private member assess in `pin::Pin::into_mode<>()`

michalfita committed:
Remove docs redundancy in `pio/pin.rs`

michalfita committed:
Update GH's actions to recent major version

michalfita committed:
Reformat code to pass formatting checks

michalfita committed:
Please `clippy`

michalfita commented:
@martinmortsell All 🟢 . Please approve.
This ended up being a lot of tiny changes to pass all our checks. But I finally managed.

martinmortsell commented on boards/atsamv71_xult/Cargo.toml:
```diff
@@ -1,6 +1,6 @@
 [package]
 name = "atsamv71_xult"
-version = "0.4.3-alpha1"
+version = "0.4.5"
```
Why are the examples at `0.4.5` but the hal at `0.4.6`?

martinmortsell reviewed (approved):
The version mismatch I found shouldn't affect anything. Fix it if you feel inclined.

Otherwise, this looks good to met.

martinmortsell commented:
Merging this should close #79.

michalfita committed:
Final touch: bump version in `atsmav71_cult` crate
